cmd/snap/inhibit.go

```diff
@@ -34,143 +34,160 @@ import (
 	"github.com/snapcore/snapd/usersession/client"
 )
 
-func waitWhileInhibited(snapName string) error {
-	hint, _, err := runinhibit.IsLocked(snapName)
-	if err != nil {
-		return err
-	}
-	if hint == runinhibit.HintNotInhibited {
-		return nil
+var runinhibitWaitWhileInhibited = runinhibit.WaitWhileInhibited
+
+func waitWhileInhibited(snapName string) (err error) {
```

Comment:
I like this, thanks for pointing out `signal.NotifyContext`. I think it might be a good idea to that in the bigger PR as mentioned here https://github.com/snapcore/snapd/pull/13467#discussion_r1458832133 since `waitWhileInhibited` might be removed later, this PR was just extracted from https://github.com/snapcore/snapd/pull/13411 as a refactor to make it more manageable.